pkg/services/dashboards/service/integration_test/dashboard_k8s_test.go

```diff
@@ -0,0 +1,1680 @@
+			require.NotNil(t, updatedDash)
+
+			// Check that version was incremented
+			updatedSpec := updatedDash.Object["spec"].(map[string]interface{})
+			updatedVersion, ok := updatedSpec["version"].(int64)
+			require.True(t, ok, "Version should be a number")
+			require.Greater(t, updatedVersion, initialVersion, "Version should be incremented after update")
+
+			// Clean up
+			err = adminClient.Resource.Delete(context.Background(), dashUID, v1.DeleteOptions{})
+			require.NoError(t, err)
+		})
+
+		// Test version conflict when updating concurrently
+		t.Run("reject update with version conflict", func(t *testing.T) {
+			// Create a dashboard with admin
+			dash, err := createDashboard(t, adminClient, "Dashboard for Version Conflict Test", nil, nil)
+			require.NoError(t, err, "Failed to create dashboard for version conflict test")
+			dashUID := dash.GetName()
+
+			// Get the dashboard twice (simulating two users getting it)
+			dash1, err := adminClient.Resource.Get(context.Background(), dashUID, v1.GetOptions{})
+			require.NoError(t, err)
+			dash2, err := editorClient.Resource.Get(context.Background(), dashUID, v1.GetOptions{})
+			require.NoError(t, err)
+
+			// Update with the first copy
+			updatedDash1, err := updateDashboard(t, adminClient, dash1, "Updated by first user")
+			require.NoError(t, err)
+			require.NotNil(t, updatedDash1)
+
+			// Try to update with the second copy (should fail with version conflict)
+			_, err = updateDashboard(t, editorClient, dash2, "Updated by second user")
+			require.Error(t, err)
+			require.Contains(t, err.Error(), "the object has been modified", "Should fail with version conflict error")
+
+			// Clean up
+			err = adminClient.Resource.Delete(context.Background(), dashUID, v1.DeleteOptions{})
+			require.NoError(t, err)
+		})
+
+		// Test setting an explicit version
+		t.Run("explicit version setting is validated", func(t *testing.T) {
+			// Create a dashboard with a specific version
+			dashObj := createDashboardObject("Dashboard with Explicit Version", "", 0)
+
+			// Set an explicit version in the spec
+			if spec, ok := dashObj.Object["spec"].(map[string]interface{}); ok {
+				spec["version"] = 5 // Set explicit version
+			}
+
+			// Create the dashboard
+			createdDash, err := adminClient.Resource.Create(context.Background(), dashObj, v1.CreateOptions{})
+			require.NoError(t, err)
+			dashUID := createdDash.GetName()
+
+			// Fetch the created dashboard
+			fetchedDash, err := adminClient.Resource.Get(context.Background(), dashUID, v1.GetOptions{})
+			require.NoError(t, err)
+
+			// Verify the version was handled properly
+			spec := fetchedDash.Object["spec"].(map[string]interface{})
+			version := spec["version"]
+			require.Equal(t, 5, version, "Version should be 5")
+
+			// Clean up
+			err = adminClient.Resource.Delete(context.Background(), dashUID, v1.DeleteOptions{})
+			require.NoError(t, err)
+		})
+	})
+
+	t.Run("Dashboard annotation validations", func(t *testing.T) {
```

Comment:
I don't believe we're doing annotation validations in the backend today, but could be missing something?

Reply:
Removed